tools/installpsh-mariner.sh

```diff
@@ -0,0 +1,216 @@
+#!/bin/bash
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+#Companion code for the blog https://cloudywindows.com
+#call this code direction from the web with:
+#bash <(wget -O - https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/installpsh-mariner.sh) ARGUMENTS
+#bash <(curl -s https://raw.githubusercontent.com/PowerShell/PowerShell/master/tools/installpsh-mariner.sh) <ARGUMENTS>
+
+#Usage - if you do not have the ability to run scripts directly from the web,
+#        pull all files in this repo folder and execute, this script
+#        automatically prefers local copies of sub-scripts
+
+#Completely automated install requires a root account or sudo with a password requirement
+
+#Switches
+# -includeide         - installs VSCode and VSCode PowerShell extension (only relevant to machines with desktop environment)
+# -interactivetesting - do a quick launch test of VSCode (only relevant when used with -includeide)
+# -skip-sudo-check    - use sudo without verifying its availability (this is required to run in the VSTS Hosted Linux Preview)
+# -preview            - installs the latest preview release of PowerShell side-by-side with any existing production releasesS
+
+#gitrepo paths are overrideable to run from your own fork or branch for testing or private distribution
+
+
+VERSION="1.2.0"
+gitreposubpath="PowerShell/PowerShell/master"
+gitreposcriptroot="https://raw.githubusercontent.com/$gitreposubpath/tools"
+thisinstallerdistro=mariner
+repobased=false
+gitscriptname="installpsh-mariner.psh"
+pwshlink=/usr/bin/pwsh
+
+echo
+echo "*** PowerShell Development Environment Installer $VERSION for $thisinstallerdistro"
+echo "***    Original script is at: $gitreposcriptroot/$gitscriptname"
+echo
+echo "*** Arguments used: $*"
+echo
+
+# Let's quit on interrupt of subcommands
+trap '
+  trap - INT # restore default INT handler
+  echo "Interrupted"
+  kill -s INT "$$"
+' INT
+
+#Verify The Installer Choice (for direct runs of this script)
+lowercase(){
+    echo "$1" | tr "[:upper:]" "[:lower:]"
+}
+
+OS=$(lowercase "$(uname)")
+if [ "${OS}" == "windowsnt" ]; then
+    OS=windows
+    DistroBasedOn=windows
+elif [ "${OS}" == "darwin" ]; then
+    OS=osx
+    DistroBasedOn=osx
+else
+    OS=$(uname)
```

Comment:
not sure why we don't use the lowercase OS created in line 52, but ok

Reply:
This is template code shared across many scripts.  I only added code for mariner.  I think we could redesign this, but I don't think it's needed.